Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.swift

```diff
@@ -0,0 +1,246 @@
+                        .task(id: directionGeometry) {
+                            guard let directionGeometry else { return }
+                            model.directionGraphic.geometry = directionGeometry
+                            await mapViewProxy.setViewpointGeometry(directionGeometry, padding: 100)
+                        }
+                        .disabled(model.route == nil)
+                        Spacer()
+                        
+                        Picker("Features", selection: $featuresSelection) {
+                            Text("Stops").tag(RouteFeatures.stops)
+                            Text("Barriers").tag(RouteFeatures.barriers)
+                        }
+                        .pickerStyle(.segmented)
+                        .labelsHidden()
+                        Spacer()
+                        
+                        SheetButton(title: "Settings") {
+                            List {
+                                Toggle(isOn: $routingFindsBestSequence) {
+                                    Text("Find Best Sequence")
+                                }
+                                .onChange(of: routingFindsBestSequence) { newValue in
+                                    model.routeParameters.findsBestSequence = newValue
+                                }
+                                
+                                Section {
+                                    Toggle(isOn: Binding(
+                                        get: { model.routeParameters.preservesFirstStop },
+                                        set: { model.routeParameters.preservesFirstStop = $0 }
+                                    )) {
+                                        Text("Preserve First Stop")
+                                    }
+                                    
+                                    Toggle(isOn: Binding(
+                                        get: { model.routeParameters.preservesLastStop },
+                                        set: { model.routeParameters.preservesLastStop = $0 }
+                                    )) {
+                                        Text("Preserve Last Stop")
+                                    }
+                                }
+                                .disabled(!routingFindsBestSequence)
+                            }
+                        } label: {
+                            Image(systemName: "gear")
+                        }
+                        Spacer()
+                        
+                        Button {
+                            model.reset(features: featuresSelection)
+                        } label: {
+                            Image(systemName: "trash")
+                        }
+                        .disabled(
+                            featuresSelection == .stops ? model.stopsCount == 0 :
+                                model.barriersCount == 0
+                        )
+                    }
```

Comment:
Not necessary, but if you wanted you could break this out into a separate view. You could pass context through the environment